pkg/controller/daemon/daemon_controller_test.go

```diff
@@ -2460,62 +2454,48 @@ func TestUpdateNode(t *testing.T) {
 				ds.Spec.Template.Spec = resourcePodSpecWithoutNodeName("200M", "200m")
 				return ds
 			}(),
-			expectedEventsFunc: func(strategyType apps.DaemonSetUpdateStrategyType) int {
-				switch strategyType {
-				case apps.OnDeleteDaemonSetStrategyType:
-					return 0
-				case apps.RollingUpdateDaemonSetStrategyType:
-					return 0
-				default:
-					t.Fatalf("unexpected UpdateStrategy %+v", strategyType)
-				}
-				return 0
-			},
 			shouldEnqueue: false,
-			expectedCreates: func() int {
-				return 1
-			},
 		},
 	}
 	for _, c := range cases {
-		for _, strategy := range updateStrategies() {
+		t.Run(c.test, func(t *testing.T) {
 			logger, ctx := ktesting.NewTestContext(t)
-			manager, podControl, _, err := newTestController(ctx)
+			manager, _, _, err := newTestController(ctx)
 			if err != nil {
 				t.Fatalf("error creating DaemonSets controller: %v", err)
 			}
-			err = manager.nodeStore.Add(c.oldNode)
-			if err != nil {
-				t.Fatal(err)
-			}
-			c.ds.Spec.UpdateStrategy = *strategy
+
 			err = manager.dsStore.Add(c.ds)
 			if err != nil {
-				t.Fatal(err)
+				t.Fatalf("error adding DaemonSet: %v", err)
 			}
 
-			expectedEvents := 0
-			if c.expectedEventsFunc != nil {
-				expectedEvents = c.expectedEventsFunc(strategy.Type)
-			}
-			expectedCreates := 0
-			if c.expectedCreates != nil {
-				expectedCreates = c.expectedCreates()
+			err = manager.nodeStore.Add(c.oldNode)
+			if err != nil {
+				t.Fatalf("error adding Node: %v", err)
 			}
-			expectSyncDaemonSets(t, manager, c.ds, podControl, expectedCreates, 0, expectedEvents)
```

Comment:
This part actually we can leave here, I think.

Reply:
Either bring this back, or implement a similar functionality. If you check the contents of `expectSyncDaemonSets` it verifies pod creations, deletions, recorder events, and controller refs. Your change drops all of those checks in favor of just verifying the queue contents. I'd rather see expanded list of checks, not a removal here.

Reply:
Done.